huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_linkage_rule_test.go

```diff
@@ -0,0 +1,257 @@
+%s
+
+resource "huaweicloud_smn_topic" "topic" {
+  name = "%s"
+}
+
+resource "huaweicloud_iotda_device_linkage_rule" "test" {
+  name     = "%s"
+  space_id = huaweicloud_iotda_space.test.id
+
+  triggers {
+    type = "DEVICE_DATA"
+    device_data_condition {
+      product_id            = huaweicloud_iotda_device.test.product_id
+      path                  = "service_1/p_1"
+      operator              = "="
+      value                 = 4
+      trigger_strategy      = "pulse"
+      data_validatiy_period = 300
+    }
+  }
+
+  actions {
+    type = "SMN_FORWARDING"
+    smn_forwarding {
+      region          = huaweicloud_smn_topic.topic.region
+      project_id      = "%s"
+      topic_name      = huaweicloud_smn_topic.topic.name
+      topic_urn       = huaweicloud_smn_topic.topic.topic_urn
+      message_title   = "title"
+      message_content = "content"
+    }
+  }
+
+  depends_on = [
+    huaweicloud_iotda_device.test,
+    huaweicloud_iotda_product.test,
+  ]
+}
+`, deviceConfig, name, name, projectId)
+}
+
+func testDeviceLinkageRule_timer(name, projectId string) string {
+	deviceConfig := testDevice_basic(name, name)
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_smn_topic" "topic" {
+  name = "%s"
+}
+
+resource "huaweicloud_iotda_device_linkage_rule" "test" {
+  name     = "%s"
+  space_id = huaweicloud_iotda_space.test.id
+  enabled  = false
+
+  triggers {
+    type = "SIMPLE_TIMER"
+    simple_timer_condition {
+      start_time      = "20220622T160000Z"
+      repeat_interval = 2
+      repeat_count    = 2
+    }
+  }
+
+  actions {
+    type = "SMN_FORWARDING"
+    smn_forwarding {
+      region          = huaweicloud_smn_topic.topic.region
+      project_id      = "%s"
+      topic_name      = huaweicloud_smn_topic.topic.name
+      topic_urn       = huaweicloud_smn_topic.topic.topic_urn
+      message_title   = "title"
+      message_content = "content"
+    }
+  }
+
+  depends_on = [
+    huaweicloud_iotda_device.test,
+    huaweicloud_iotda_product.test,
+  ]
+}
+`, deviceConfig, name, name, projectId)
+}
+
+func testDeviceLinkageRule_daily(name string) string {
+	deviceConfig := testDevice_basic(name, name)
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_smn_topic" "topic" {
+  name = "%s"
+}
+
+resource "huaweicloud_iotda_device_linkage_rule" "test" {
+  name     = "%s"
+  space_id = huaweicloud_iotda_space.test.id
+  enabled  = false
+
+  triggers {
+    type = "DAILY_TIMER"
+    daily_timer_condition {
+      start_time   = "19:02"
+      days_of_week = "1,2,3,4,5,6,7"
+    }
+  }
+
+  actions {
+    type = "DEVICE_CMD"
+    device_command {
+      device_id    = huaweicloud_iotda_device.test.id
+      service_id   = "service_1"
+      command_name = "cmd_1"
+      command_body = "{\"cmd_p_1\":\"3\"}"
+    }
+  }
+
+  effective_period {
+    start_time   = "00:00"
+    end_time     = "23:59"
+    days_of_week = "1,2,3,4,5,6"
+  }
+
+  depends_on = [
+    huaweicloud_iotda_device.test,
```

Comment:
do we need this depends_on? seems you already do reference to it at L238

Reply:
There are 3 test cases,  prevent the device from being deleted